lib/puppet/type/acl/ace.rb

```diff
@@ -124,12 +164,18 @@ def ensure_rights_values_compatible
       end
     end
 
+    # Ensures that `mask` is set to `value` when `rights` is set to `mask_specific`.
+    # An error is raised if the condition is not matched.
     def ensure_mask_when_mask_specific
       if @rights.include?(:mask_specific) && (@mask.nil? || @mask.empty?) # rubocop:disable Style/GuardClause  Changing this to a guard clause makes the line long and unreadable
         raise ArgumentError, "If you specify rights => ['mask_specific'], you must also include mask => 'value'. Reference: #{inspect}"
       end
     end
 
+    # Checks that a supplied array of values are unique
+    #
+    # @param [Array] values
+    # @return [Boolean] If values are unique or not
```

Comment:
I don't think the return here is going to be a bool. It's either going to return a new array with only unique values, or if it's not an array return to supplied value.